src/actions/Deselect.ts

```diff
@@ -0,0 +1,41 @@
+import { Selection } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  TypedSelection,
+  Graph,
+} from "../typings/Types";
+import { runOnTargetsForEachEditor } from "../util/targetUtils";
+
+export default class Deselect implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await runOnTargetsForEachEditor(targets, async (editor, targets) => {
+      // Remove selections with a non-empty intersection
+      const newSelections = editor.selections.filter(
+        (selection) =>
+          !targets.find((target) => {
+            const intersection =
+              target.selection.selection.intersection(selection);
+            return intersection && !intersection.isEmpty;
+          })
+      );
+      // The editor requires at least one selection. Keep "primary" selection active
```

Comment:
Shouldn't we just let this be an error?

Reply:
We could if you prefer that. If you have a really large selection just getting that out of the way with that the select command might be nice. It basically turns into an escape in that case

Reply:
Oh that's kinda cool i guess.  Feels logically incorrect but useful enough to let it slide 😊